StudentList/Fragments/StudentProfileFragment.cs

```diff
@@ -116,7 +121,7 @@ public override bool OnOptionsItemSelected(IMenuItem item)
             switch (item.ItemId)
             {
                 case Resource.Id.action_confirm:
-                    this.ConfirmAsync();
+                    this.Activity.RunMethodWithLoaderAsync(this.ConfirmAsync());
```

Comment:
Same here

Reply:
But I have a several async/await methods in ConfirmAsync method and LoadingDialog should be shown when they are running. Currently, I have the one solving, I can just write smth that:
private async Task ConfirmAsync()
{ 
     loadingDialog.Show();
     //async/await method running
     loadingDialog.Hide()
}

So I don't know if this is correct. Do you have any ideas?

Reply:
The main idea is to wrap exactly a thing that you want to actually be covered by loader. Usually, it's some single long-running operation, that's why I recommended to wrap repo calls. But of course, if you have several long-running operations and you want them all to run with a same loader, then you can have a wrapper function (like ConfirmAsync) and surround it with loader.